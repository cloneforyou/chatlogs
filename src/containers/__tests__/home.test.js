import React from 'react';
import { shallow } from 'enzyme';
import Home from '../home';

describe('<Home />', () => {
  const props = {
    messages: [],
    errorMessage: '',
    isLoading: false,
    loadMessages: () => {},
  };

  it('should render loading', () => {
    const container = shallow(
      <Home.WrappedComponent {...props} isLoading={true} />
    );
    expect(container.find('p').text()).toEqual('Loading....');
  });

  it('should display errorMessage', () => {
    const container = shallow(
      <Home.WrappedComponent
        {...props}
        errorMessage="something went wrong"
        isLoading={false}
      />
    );
    expect(container.find('p').text()).toEqual('something went wrong');
  });
  it('renders correctly', () => {
    const tree = shallow(<Home.WrappedComponent {...props} />);
    expect(tree).toMatchSnapshot();
  });
});
