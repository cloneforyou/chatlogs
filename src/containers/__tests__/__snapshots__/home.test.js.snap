// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home /> renders correctly 1`] = `
<div
  className="container"
>
  <MessageList
    messages={Array []}
  />
</div>
`;
